### PR TITLE
feat: BI-5951 presigned url generation in file-uploader-api

### PR DESCRIPTION
**Pull request by KonstantAnxiety**
Also includes:
- decrease DataFile Redis TTL from 12h to 3h
- rename all DataFile instances across the codebase to "dfile"
- transition from DataFile s3_key generated from its id to s3_key stored in Redis

### CODE REVIEW

File: lib/dl_file_uploader_api_lib/dl_file_uploader_api_lib/views/files.py [hunk 1]
```diff
@@ -104,31 +105,55 @@ async def _chunk_iter(chunk_size: int = 10 * 1024 * 1024) -> AsyncGenerator[byte
...
         )
 
 
+class MakePresignedUrlView(FileUploaderBaseView):
+    async def post(self) -> web.StreamResponse:
+        req_data = await self._load_post_request_schema_data(files_schemas.MakePresignedUrlRequestSchema)
+        content_md5: str = req_data["content_md5"]
+
+        s3 = self.dl_request.get_s3_service()
+        s3_key = "{}_{}".format(self.dl_request.rci.user_id or "unknown", str(uuid.uuid4()))
```
Comment (ovsds) [inline lib/dl_file_uploader_api_lib/dl_file_uploader_api_lib/views/files.py:139]: don't we wanna make it `{user}/{uuid}`? s3 is fs afterall, so it sounds reasonable.

File: lib/dl_file_uploader_api_lib/dl_file_uploader_api_lib/views/files.py [hunk 2]
```diff
@@ -104,31 +105,55 @@ async def _chunk_iter(chunk_size: int = 10 * 1024 * 1024) -> AsyncGenerator[byte
...
+
+        url = await s3.client.generate_presigned_post(
+            Bucket=s3.tmp_bucket_name,
+            Key=s3_key,
+            ExpiresIn=60 * 60,  # 1 hour
```
Comment (ovsds) [inline lib/dl_file_uploader_api_lib/dl_file_uploader_api_lib/views/files.py:144]: magic numbers to CONSTS or ClassVar?

File: lib/dl_file_uploader_api_lib/dl_file_uploader_api_lib/views/files.py [hunk 3]
```diff
@@ -82,13 +83,13 @@ async def post(self) -> web.StreamResponse:
         s3 = self.dl_request.get_s3_service()
 
         rmm = self.dl_request.get_redis_model_manager()
-        df = DataFile(
+        dfile = DataFile(
```
Comment (ovsds) [inline lib/dl_file_uploader_api_lib/dl_file_uploader_api_lib/views/files.py:86]: nitpick: data_file?

File: lib/dl_file_uploader_lib/dl_file_uploader_lib/redis_model/models/models.py [hunk 4]
```diff
@@ -195,10 +196,14 @@ class DataFile(RedisModelUserIdAuth):
...
+    DEFAULT_TTL: ClassVar[Optional[int]] = 3 * 60 * 60  # 3 hours
 
     @property
-    def s3_key(self) -> str:
+    def s3_key_old(self) -> str:
+        # transition from s3_key generated by self.id to stored self.s3_key, to be removed in future releases
+        # see also: DataFileSchema
+        if self.s3_key is not None:
+            return self.s3_key
         return self.id
```
Comment (ovsds) [inline lib/dl_file_uploader_lib/dl_file_uploader_lib/redis_model/models/models.py:207]:
```suggestion
        return self.s3_key or self.id
```

File: lib/dl_file_uploader_lib/dl_file_uploader_lib/redis_model/models/models.py [hunk 5]
```diff
@@ -195,10 +196,14 @@ class DataFile(RedisModelUserIdAuth):
     error: Optional[FileProcessingError] = attr.ib(default=None)
 
     KEY_PREFIX: ClassVar[str] = "df"
-    DEFAULT_TTL: ClassVar[Optional[int]] = 12 * 60 * 60  # 12 hours
+    DEFAULT_TTL: ClassVar[Optional[int]] = 3 * 60 * 60  # 3 hours
 
     @property
-    def s3_key(self) -> str:
```
Comment (ovsds) [inline lib/dl_file_uploader_lib/dl_file_uploader_lib/redis_model/models/models.py:201]: nitpick: sound's like it was easier to make `s3_key_new` field and rename it later to `s3_key`. would be way less code changes

File: lib/dl_s3/dl_s3/s3_service.py [hunk 6]
```diff
@@ -59,6 +60,7 @@ async def initialize(self) -> None:
             aws_access_key_id=self._access_key_id,
             aws_secret_access_key=self._secret_access_key,
             endpoint_url=self._endpoint_url,
+            config=AioConfig(signature_version="s3v4"),
```
Comment (ovsds) [inline lib/dl_s3/dl_s3/s3_service.py:63]: why and what is that?

**Review by ovsds — APPROVED** [review]

**Comment by KonstantAnxiety** [comment]
Going to include fixes for comments in the 2nd part of this PR, since I've already pushed a part of it https://github.com/datalens-tech/datalens-backend/pull/762

**Pull request merged**